app/src/main/java/com/github/bgabriel998/softwaredevproject/general/SettingsMapActivity.java

```diff
@@ -0,0 +1,178 @@
+package com.github.bgabriel998.softwaredevproject.general;
+
+import android.annotation.SuppressLint;
+import android.app.Activity;
+import android.content.Context;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.core.util.Pair;
+
+import com.github.bgabriel998.softwaredevproject.R;
+import com.github.bgabriel998.softwaredevproject.map.OSMMap;
+import com.github.bgabriel998.softwaredevproject.points.DownloadTopographyTask;
+import com.github.bgabriel998.softwaredevproject.points.GeonamesHandler;
+import com.github.bgabriel998.softwaredevproject.points.POIPoint;
+import com.github.bgabriel998.softwaredevproject.points.Point;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.osmdroid.bonuspack.location.POI;
+
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Activity that allows the user to select a point around which compute and
+ * download POIPoints and an elevation map for offline usage.
+ */
+public class SettingsMapActivity extends AppCompatActivity {
+
+    private OSMMap osmMap = null;
+
+    private Button backButton;
+    private Button okButton;
+
+    Activity thisActivity;
+
+    private Point selectedPoint;
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_settings_map);
+
+        this.backButton = findViewById(R.id.settingsMapBackButton);
+        this.backButton.setOnClickListener(v -> this.finish());
+
+        this.okButton = findViewById(R.id.settingsMapOkButton);
+        this.okButton.setOnClickListener(v -> {
+            saveToJson();
+        });
```

Comment:
Can be replaced with lambda.
```suggestion
        this.okButton.setOnClickListener(v -> saveToJson());
```